Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -731,6 +731,29 @@ static _WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
     }
 }
 
+bool WebExtensionContext::hasPermissions(PermissionsSet permissions, MatchPatternSet matchPatterns)
+{
+    for (auto& permission : permissions) {
+        if (!m_grantedPermissions.contains(permission))
+            return false;
+    }
+
+    for (auto& pattern : matchPatterns) {
+        bool matchFound = false;
+        for (auto& grantedPattern : currentPermissionMatchPatterns()) {
+            if (pattern->matchesPattern(grantedPattern)) {
```

Comment:
This needs to pass the `{ WebExtensionMatchPattern::Options::IgnorePaths }` option to `matchesPattern()`.